pkg/plugins/policies/meshproxypatch/plugin/v1alpha1/cluster_mod.go

```diff
@@ -0,0 +1,71 @@
+package v1alpha1
+
+import (
+	envoy_cluster "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	envoy_resource "github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+	protov1 "github.com/golang/protobuf/proto"
+	"github.com/pkg/errors"
+
+	core_xds "github.com/kumahq/kuma/pkg/core/xds"
+	api "github.com/kumahq/kuma/pkg/plugins/policies/meshproxypatch/api/v1alpha1"
+	util_proto "github.com/kumahq/kuma/pkg/util/proto"
+)
+
+type clusterModificator api.ClusterMod
+
+func (c *clusterModificator) apply(resources *core_xds.ResourceSet) error {
+	clusterMod := &envoy_cluster.Cluster{}
+	if c.Value != nil {
+		if err := util_proto.FromYAML([]byte(*c.Value), clusterMod); err != nil {
+			return err
+		}
+	}
+	switch c.Operation {
+	case api.ModOpAdd:
+		c.add(resources, clusterMod)
+	case api.ModOpRemove:
+		c.remove(resources)
+	case api.ModOpPatch:
+		c.patch(resources, clusterMod)
+	default:
+		return errors.Errorf("invalid operation: %s", c.Operation)
+	}
+	return nil
+}
+
+func (c *clusterModificator) patch(resources *core_xds.ResourceSet, clusterMod *envoy_cluster.Cluster) {
+	for _, cluster := range resources.Resources(envoy_resource.ClusterType) {
+		if c.clusterMatches(cluster) {
+			util_proto.Merge(protov1.MessageV2(cluster.Resource), clusterMod)
```

Comment:
Can we even have MessageV1 in `Resource`? I think Envoy proto messages have a v2 version, isn't it?